### PR TITLE
Empty Stack

### PR DESCRIPTION
**Pull request by JohnCoene**
This initialises work on allowing for an empty stack.

@DivadNojnarg it admittedly requires some refactoring but let me know what you think.

### CODE REVIEW

File: R/stack.R [hunk 1]
```diff
@@ -61,32 +65,40 @@ generate_code.stack <- function(x) {
 #' @return Invisibly returns the stack.
 #' @export
 add_block <- function(stack, block, position = NULL) {
-  stopifnot(length(stack) > 0)
   if (is.null(position)) stopifnot(position >= 1)
```
Comment (DivadNojnarg) [inline R/stack.R:68]: I believe we should remove this line.

**Comment by JohnCoene** [comment]
Sorry for being pushy but could we look at this PR, I really need to be able to initialise empty stacks in order to proceed with other tasks.

**Comment by DivadNojnarg** [comment]
I am trying to test it but something is broken. Now the passed block to `add_block` is a function

```r
$ block
function(...) {
  initialize_block(new_data_block(...))
}
```

 and not a structure which crashes on the first check:

```r
Error in add_block: The first block must be a data block.
```

File: R/stack.R [hunk 2]
```diff
@@ -61,32 +65,40 @@ generate_code.stack <- function(x) {
...
   if (length(stack) == 0) {
-    block_name <- deparse(substitute(block))
```
Comment (DivadNojnarg) [inline R/stack.R:67]: This is what causes the issue with the block class check because block is a function.

**Comment by JohnCoene** [comment]
That's what I get from @nbenn registry `available_blocks()`, e.g.: `new_block <- reactive({available_blocks()[1}[[1]])`

How should I go about it?
It's confusing because the registry doesn't return a list of blocks, these are `block_descr`

File: R/stack.R (hunk 2 above)
Comment (JohnCoene) [reply R/stack.R:67]: How do you use/test this, could you share an example?

Comment (DivadNojnarg) [reply R/stack.R:67]: @JohnCoene I just pushed a fix for the example. I noticed another issue, but this is actually not related to this PR. It seems that blocks in the registry don't call the `initialize_block` (`new_data_block` and `data_block` are different, the latter calling `initialize_block(new_data_block(...))`) method that's why you get an error why you insert a filter block. This is explains #172 and #165.

**Review by DivadNojnarg — APPROVED** [review]
LGTM.

File: R/stack.R (hunk 2 above)
Comment (JohnCoene) [reply R/stack.R:67]: Ah good find!

**Comment by JohnCoene** [comment]
Thanks David!

**Pull request merged**